### ISSUE TITLE
No tunnel styling for some railway types

### ISSUE THREAD

**Issue by wmyrda** [openstreetmap-carto/openstreetmap-carto#3168]
Railway track located in tunnel should be distinguishable from regular over ground tracks. 

I did not follow carto development all that closely, but when I last edited this object https://www.openstreetmap.org/way/225995257 it was clear that it is located in the tunnel. Now I am unable to see any difference if there is any at all.
I would except map would treat underground object somewhat differently.

**Comment by kocio-pl** [comment]
I'm not a railway mapper, but we have a lot of such rendering - look here for example (maybe that's because of layer<0 ?):

https://www.openstreetmap.org/way/139150499#map=18/52.22755/20.99758

**Comment by lakedistrictOSM** [comment]
@kocio-pl your example is `railway=rail` which renders as a dashed line when tagged with `tunnel=yes`.

@wmyrda's example is `railway=narrow_gauge`. `railway=preserved` also doesn't render as a dashed line - https://www.openstreetmap.org/way/87193371

**Comment by wmyrda** [comment]
For a second my believe was there was rendering change due to someone finding a case where tunnels may be located over the ground and layer is now required so I've added layer=-1 to my example to see if that might be the case, but seems it did not help as tunnel is still not shown.

**Comment by matthijsmelissen** [comment]
This indeed seems to be a regression: http://bl.ocks.org/matthijsmelissen/raw/328d6a95a431996813ece839aaf9ff7b/#18.00/49.88225/22.24578

**Comment by dch0ph** [comment]
@imagico I've been looking through the open railway issues. I'm pretty sure this is resolved - from your samples in #5290, there are no railway types where tunnel tagging is ignored.

**Comment by imagico** [comment]
As tests on #4965 showed the only remaining railway type without a distinct tunnel rendering is now `railway=miniature` - about 1000 uses with `tunnel`/`covered`.

**Comment by dch0ph** [comment]
True, but we now have the specific issue #5191 for tunnels on miniature railways.

**Comment by imagico** [comment]
Good point, closing this.
